api/client_test.go

```diff
@@ -1,71 +1,92 @@
 package api
 
 import (
+	"math/rand"
 	"os"
 	"strconv"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func TestDefaultClientConfig_WithOSEnvSet(t *testing.T) {
+func Test_DefaultClientConfig_WithEnvVars(t *testing.T) {
+	t.Cleanup(func() {
+		_ = os.Unsetenv(EnvAddress)
+		_ = os.Unsetenv(EnvTLSCACert)
+		_ = os.Unsetenv(EnvTLSCAPath)
+		_ = os.Unsetenv(EnvTLSClientCert)
+		_ = os.Unsetenv(EnvTLSClientKey)
+		_ = os.Unsetenv(EnvTLSSSLVerify)
+	})
+
 	url := "https://1.2.3.4:5678"
 	caCert := "test/path/ca.pem"
 	caPath := "test/path"
 	clientCert := "test/path/client.pem"
 	clientKey := "test/path/key.pem"
 	sslVerify := "false"
 
-	err := os.Setenv(EnvAddress, url)
-	require.NoError(t, err)
-	defer os.Setenv(EnvAddress, "")
-
-	err = os.Setenv(EnvTLSCACert, caCert)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSCACert, "")
-
-	err = os.Setenv(EnvTLSCAPath, caPath)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSCAPath, "")
-
-	err = os.Setenv(EnvTLSClientCert, clientCert)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSClientCert, "")
-
-	err = os.Setenv(EnvTLSClientKey, clientKey)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSClientKey, "")
-
-	os.Setenv(EnvTLSSSLVerify, sslVerify)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSSSLVerify, "")
+	require.NoError(t, os.Setenv(EnvAddress, url))
+	require.NoError(t, os.Setenv(EnvTLSCACert, caCert))
+	require.NoError(t, os.Setenv(EnvTLSCAPath, caPath))
+	require.NoError(t, os.Setenv(EnvTLSClientCert, clientCert))
+	require.NoError(t, os.Setenv(EnvTLSClientKey, clientKey))
+	require.NoError(t, os.Setenv(EnvTLSSSLVerify, sslVerify))
 
-	config := DefaultClientConfig()
+	clientConfig := DefaultClientConfig()
 
-	assert.Equal(t, url, config.Addr)
-	assert.Equal(t, caCert, config.TLSConfig.CACert)
-	assert.Equal(t, caPath, config.TLSConfig.CAPath)
-	assert.Equal(t, clientCert, config.TLSConfig.ClientCert)
-	assert.Equal(t, clientKey, config.TLSConfig.ClientKey)
+	require.Equal(t, url, clientConfig.Addr)
+	require.Equal(t, caCert, clientConfig.TLSConfig.CACert)
+	require.Equal(t, caPath, clientConfig.TLSConfig.CAPath)
+	require.Equal(t, clientCert, clientConfig.TLSConfig.ClientCert)
+	require.Equal(t, clientKey, clientConfig.TLSConfig.ClientKey)
```

Comment:
Curious to learn about the reason to unify on 'require'. I know that some of the mix of 'require' and 'assert' is done deliberately. Here, I think  'assert' was used because if one Equal check fails, it's still worth continuing to check the others. The earlier 'require's are likely because they check test setup - if the test setup fails, then its not worth continuing the test.

Reply:
Ah thanks for explaining that, I wasn't aware that using `require` implies 'fail fast'. I thought the behavior was the same and we were getting it from two different packages so I wanted to unify that.
I will switch back to using asserts in places we don't need to fail fast.

Reply:
Addressed in https://github.com/hashicorp/consul-terraform-sync/pull/663/commits/63d9a6d23abf5cec7b5928f73d6f56da5836fbcb

Reply:
Gotcha, thanks for clarifying! Sounds good.